chatkit-android/src/main/kotlin/com/pusher/chatkit/BeamsPushNotifications.kt

```diff
@@ -38,7 +32,25 @@ class BeamsPushNotifications(
     override fun setUserId(userId: String): Result<Unit, Error> {
         val f = FutureValue<Result<Unit, Error>>()
         try {
-            Beams.setUserId(userId, object : Callback<Void, PusherCallbackError> {
+            val tokenProvider = object : TokenProvider {
+                override fun fetchToken(userId: String): String {
+                    val token = beamsTokenProviderService.fetchToken(userId).map { it.token }
+                    when(token) {
```

Comment:
it is a bit confusing that we are calling function that returns a `Result` type then force it to be `String` and then back into a `Result` type. I think it would make the code a bit cleaner to map over `Result` one more time. I haven't checked if this works, but something like 
```kotlin
beamsTokenProviderService
  .fetchToken(userId)
  .map { it.token }
  .map { it.value }
  .recover {throw ...}
```

Reply:
I offered something more similar to that in #110